### PR TITLE
Included shellCheck in travis build #95

### LINKED ISSUE #95: Include shellCheck in travis builds
## Is this a BUG REPORT or FEATURE REQUEST?
 FEATURE REQUEST

- Multiple images in the test-tools repo make use of bash scripts as the container business logic. They need to be validated via shellCheck utility as part of sanity checks before building/pushing the images. 

Ref: https://dev.to/david_j_eddy/using-shellcheck-to-lint-your-bashsh-scripts-3jaf

### PR DESCRIPTION
**Pull request by biswaroop1547**
**What this PR does / why we need it**:
Validated via shellCheck utility as part of sanity checks for bash scripts before building/pushing the docker images in travis build
**Which issue this PR fixes**: 
fixes #95 

**Special notes for your reviewer**:
checks for bash scripts recursively inside each folder and does shellcheck respectively in travis build.

### CODE REVIEW

**Comment by biswaroop1547** [comment]
some checks fails because of the errors founded by doing shellcheck on the bash script files

File: .travis.yml [hunk 1]
```diff
@@ -10,91 +10,119 @@ before_install:
...
...bench-runner.sh tpcc-client/tpcc-mysql/tpcc-runner.sh tpcc-client/tpcc-mysql/load.sh tpcc-client/tpcc-mysql/load_multi_schema.sh tpcc-client/tpcc-mysql/scripts/analyzeR.sh tpcc-client/tpcc-mysql/scripts/runX.sh tpcc-client/tpcc-mysql/scripts/2instances/run.sh tpcc-client/tpcc-mysql/scripts/innodb_stat.sh tpcc-client/tpcc-mysql/scripts/analyze_flushed.sh tpcc-client/tpcc-mysql/scripts/run.sh tpcc-client/tpcc-mysql/scripts/analyze_evicted.sh tpcc-client/tpcc-mysql/scripts/remote/runX.sh tpcc-client/tpcc-mysql/scripts/run_no_backup.sh tpcc-client/tpcc-mysql/scripts/analyze_checkpoint_xtradb.sh tpcc-client/tpcc-mysql/scripts/analyze_min.sh tpcc-client/tpcc-mysql/scripts/analyze_dirty.sh tpcc-client/tpcc-mysql/scripts/virident_stat.sh tpcc-client/tpcc-mysql/scripts/analyze_checkpoint.sh tpcc-client/tpcc-mysql/scripts/multi-instances/runX.sh tpcc-client/tpcc-mysql/scripts/analyze_checkpoint_xtradb1.sh tpcc-client/tpcc-mysql/scripts/analyze_modified.sh tpcc-client/tpcc-mysql/scripts/analyze_flushed_innodb.sh tpcc-client/tpcc-mysql/scripts/flashcache_stat.sh tpcc-client/tpcc-mysql/scripts/analyze.full.sh tpcc-client/tpcc-mysql/scripts/analyze.sh vdbench/vdbench.bash vdbench/bench_runner.sh
```
Comment (ksatchit) [inline .travis.yml]: Can we obtain a list of bash scripts via a bash command and iterate over that instead of listing them manually here?

File: .travis.yml [hunk 2]
```diff
@@ -10,91 +10,119 @@ before_install:
...
+
+
 jobs:
   include:
     - stage: build docker images
       name: build docker image vdbench
       script:
+      # - shellcheck $(find ./vdbench -type f ! -name 'push' | grep .sh)
```
Comment (ksatchit) [inline .travis.yml]: We can remove the commented lines!

File: busybox/liveness.sh [hunk 3]
```diff
@@ -16,7 +16,7 @@ if [ "$?" == 0 ]; then
   echo "liveness-running"
   sleep $LIVENESS_TIMEOUT_SECONDS
 else
-  echo "livenes-failed"
+  echo "liveness-failed"
```
Comment (ksatchit) [inline busybox/liveness.sh]: :+1:

**Comment by ksatchit** [comment]
@biswaroop1547 , thank you for the PR! Have a few minor comments.

File: .travis.yml (hunk 1 above)
Comment (biswaroop1547) [reply .travis.yml]: Okay Sure! I'll do it

File: .travis.yml [hunk 4]
```diff
@@ -10,91 +10,119 @@ before_install:
   - sleep 15
   - sudo apt-get install -y
   - sudo apt-get install -y curl
+  - sudo apt-get install -y shellcheck
+
+language: shell
+  script:
+    - bash -c 'shopt -s globstar nullglob; shellcheck **/*.{sh,ksh,bash}'
```
Comment (shubhamchaudhary) [inline .travis.yml]: ```shellcheck **/*.{sh,ksh,bash}``` will lookup for scripts  in all directories and check their syntax . Why did you add an extra line (line 18) to list the scripts..?

Comment (biswaroop1547) [reply .travis.yml]: As when I was looking about shellcheck with Travis , found this https://github.com/koalaman/shellcheck/wiki/TravisCI and I was unsure about globstar and nullglob so for confirmation I added the extra line for all the scripts , think I should remove line 18 , as line 17 is already doing shellcheck on all the scripts.

Comment (shubhamchaudhary) [reply .travis.yml]: Can you please check the indentation , I think script should be sibling of language not its child?

File: .travis.yml [hunk 5]
```diff
@@ -13,42 +13,36 @@ before_install:
...
...bench-runner.sh tpcc-client/tpcc-mysql/tpcc-runner.sh tpcc-client/tpcc-mysql/load.sh tpcc-client/tpcc-mysql/load_multi_schema.sh tpcc-client/tpcc-mysql/scripts/analyzeR.sh tpcc-client/tpcc-mysql/scripts/runX.sh tpcc-client/tpcc-mysql/scripts/2instances/run.sh tpcc-client/tpcc-mysql/scripts/innodb_stat.sh tpcc-client/tpcc-mysql/scripts/analyze_flushed.sh tpcc-client/tpcc-mysql/scripts/run.sh tpcc-client/tpcc-mysql/scripts/analyze_evicted.sh tpcc-client/tpcc-mysql/scripts/remote/runX.sh tpcc-client/tpcc-mysql/scripts/run_no_backup.sh tpcc-client/tpcc-mysql/scripts/analyze_checkpoint_xtradb.sh tpcc-client/tpcc-mysql/scripts/analyze_min.sh tpcc-client/tpcc-mysql/scripts/analyze_dirty.sh tpcc-client/tpcc-mysql/scripts/virident_stat.sh tpcc-client/tpcc-mysql/scripts/analyze_checkpoint.sh tpcc-client/tpcc-mysql/scripts/multi-instances/runX.sh tpcc-client/tpcc-mysql/scripts/analyze_checkpoint_xtradb1.sh tpcc-client/tpcc-mysql/scripts/analyze_modified.sh tpcc-client/tpcc-mysql/scripts/analyze_flushed_innodb.sh tpcc-client/tpcc-mysql/scripts/flashcache_stat.sh tpcc-client/tpcc-mysql/scripts/analyze.full.sh tpcc-client/tpcc-mysql/scripts/analyze.sh vdbench/vdbench.bash vdbench/bench_runner.sh
```
Comment (shubhamchaudhary) [inline .travis.yml]: Can you remove this comment also?

File: .travis.yml (hunk 4 above)
Comment (shubhamchaudhary) [reply .travis.yml]: In scripts, the standard way is to provide only 2 spaces in the beginning of the child block. Can you please provide only 2 spaces in beginning of line 17.

**Comment by biswaroop1547** [comment]
why the DCO is failing .. i have no idea even tho i signed my commits

**Comment by biswaroop1547** [comment]
Why the DCO is failing .. even though I have signed my commits ..

**Comment by biswaroop1547** [comment]
@ksatchit @shubhamchaudhary please check

**Comment by shubhamchaudhary** [comment]
> @ksatchit @shubhamchaudhary please check



> @ksatchit @shubhamchaudhary please check

looks good , why did you force pushed the commits ?. Due to that some changes from other PR is also added in your PR so can you resolve that issue? . Please take a look to the Files changed section.

**Comment by shubhamchaudhary** [comment]
Can you please check the travis logs, the shellcheck command is not executed there.
Can you create one extra stage to run shellcheck something like that :+1: 
``` stages:
  - shell check
``` 
then include shellcheck command like 
```
 - stage: shell check
      name: shell checking
      script:
      - bash -c 'shopt -s globstar nullglob; shellcheck **/*.{sh,ksh,bash}' || true
``` 
here true is used to ignore the errors

**Comment by biswaroop1547** [comment]
please check now @shubhamchaudhary and thank you for helping me out!

**Comment by shubhamchaudhary** [comment]
Great job @biswaroop1547 , If you are really interested and want to contribute more in future. You can star our repository "github.com/litmuschaos/*"  ( here * is having same meaning  as you are using in your code  ```  shellcheck **/*.{sh,ksh,bash}' ``` )

**Comment by NisantaSahoo** [comment]
Thanks for taking part in Hacktoberfest 2019. Please fill up this form to claim your swag https://hubs.ly/H0l5ldm0
